packages/firebase_storage/firebase_storage/lib/src/utils.dart

```diff
@@ -27,7 +27,7 @@ const String _cloudStorageHost =
 const String _bucketDomain = r'([A-Za-z0-9.\-_]+)';
 const String _version = 'v[A-Za-z0-9_]+';
 const String _firebaseStoragePath = r'(/([^?#]*).*)?$';
-const String _cloudStoragePath = r'([^?#]*)*$';
+const String _cloudStoragePath = '([^?#]*)';
```

Comment:
Could you add the link to the JS code as a comment so we can find it quicker for reference?